optuna/trial/_frozen.py

```diff
@@ -546,6 +546,16 @@ def create_trial(
         Please note that this is a low-level API. In general, trials that are passed to objective
         functions are created inside :func:`~optuna.study.Study.optimize`.
 
+    .. note::
+
+        You cannot specify both ``value`` and ``values`` when ``state`` is ``None``
+        or ``TrialState.COMPLETE``.
+
+    .. note::
+
+        Both ``params`` and ``distributions`` are required when ``state``
+        is ``None`` or ``TrialState.COMPLETE``.
+
     Args:
         state:
             Trial state.
```

Comment:
How about specifying the default value here, e.g. `The default value of this argument is ``TrialState.COMPLETE``.`, because we can easily understand why we need to follow some rules when ``state`` is ``None`` or ``TrialState.COMPLETE``.

Reply:
> because we can easily understand why we need to follow some rules when `state` is `None` or `TrialState.COMPLETE`.

Sorry, but "some rules" is too ambiguous to me.

Also I don't know what you mean by "the default value",  I guess you refer `state` from the words following "e.g.". If this is the case, what you say is wrong: the default value of `state` is `None`, and Optuna regard `state` of `None` as `TrialState.COMPLETE`.

Anyway, I respect your point, but I can't entirely agree with it because the default values are obvious (see the capture below) even if the type hints are kind of annoying to me.
![image](https://user-images.githubusercontent.com/16191443/102794503-d339f380-43ee-11eb-8fe0-5259a8df0088.png)

Reply:
Oh, sorry, `TrialState.COMPLETE` is not the default value, actually. `None` is converted to `TrialState.COMPLETE`.

In that case, how about writing as follows?

```
        value:
            Trial objective value. Must be specified if ``state`` is :class:`TrialState.COMPLETE` or ``None``.
        values:
            Sequence of the trial objective values. The length is greater than 1 if the problem is
            multi-objective optimization.
            Must be specified if ``state`` is :class:`TrialState.COMPLETE` or ``None``.
```

Reply:
I'm sorry, but which part are you mentioning?

The quote block is just a copy&paste of the part of `value` and `values` of the API documentation.

Reply:
I meant replacing ":class:`TrialState.COMPLETE`" to ":class:`TrialState.COMPLETE` or ``None``".

Reply:
I'm so sorry, but I still do not understand what you say.

This is a thread on the addition of `note` block about `params` and `distributions`. But now what you suggest looks like a change on the descriptions of `value` and `values`.

Do you want me to update the descriptions of `value` and `values` in the parameters section?
Do you think the addition of `note` which I authored is not necessary here?

Reply:
> Do you want me to update the descriptions of value and values in the parameters section?

Yes. When I checked the note and the descriptions of `value` and `values`, it seems 
inconsistent for me, because the note says `Both ``params`` and ``distributions`` are required when ``state`` is None or TrialState.COMPLETE`, while the descriptions of `value` and `values` say `Must be specified if ``state`` is TrialState.COMPLETE`.

Reply:
> Do you want me to update the descriptions of value and values in the parameters section?

Yes. When I checked the note and the descriptions of value and values, it seems
inconsistent for me, because the note says `You cannot specify both ``value`` and ``values`` when ``state`` is ``None`` or ``TrialState.COMPLETE`` ` by specifying ` ``None`` or ``TrialState.COMPLETE`` `, while the descriptions of value and values say `Must be specified if ``state`` is TrialState.COMPLETE ` by specifying only ` ``TrialState.COMPLETE`` ` .